File: Final/erp-backend/src/main/java/com/pyme/erp/service/ProveedorService.java
package com.pyme.erp.service;

import com.pyme.erp.dto.request.ProveedorRequest;
import com.pyme.erp.model.Proveedor;

import java.util.List;

public interface ProveedorService {
    
    Proveedor crear(ProveedorRequest request);
    
    Proveedor actualizar(Long id, ProveedorRequest request);
    
    Proveedor obtenerPorId(Long id);
    
    List<Proveedor> listarActivos();
    
    void desactivar(Long id);
    
    void activar(Long id);
    
    List<Proveedor> buscarPorNombre(String nombre);
}

File: Final/erp-frontend/src/app/services/proveedor.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ResponseWrapper, Proveedor } from '../models/response-wrapper.model';

@Injectable({
  providedIn: 'root'
})
export class ProveedorService {
  private baseUrl = 'http://localhost:8080/api/proveedores';

  constructor(private http: HttpClient) { }

  crear(proveedor: Proveedor): Observable<ResponseWrapper<Proveedor>> {
    return this.http.post<ResponseWrapper<Proveedor>>(this.baseUrl, proveedor);
  }

  listar(): Observable<ResponseWrapper<Proveedor[]>> {
    return this.http.get<ResponseWrapper<Proveedor[]>>(this.baseUrl);
  }

  obtenerPorId(id: number): Observable<ResponseWrapper<Proveedor>> {
    return this.http.get<ResponseWrapper<Proveedor>>(`${this.baseUrl}/${id}`);
  }

  actualizar(id: number, proveedor: Proveedor): Observable<ResponseWrapper<Proveedor>> {
    return this.http.put<ResponseWrapper<Proveedor>>(`${this.baseUrl}/${id}`, proveedor);
  }

  desactivar(id: number): Observable<ResponseWrapper<void>> {
    return this.http.delete<ResponseWrapper<void>>(`${this.baseUrl}/${id}`);
  }

  activar(id: number): Observable<ResponseWrapper<void>> {
    return this.http.put<ResponseWrapper<void>>(`${this.baseUrl}/${id}/activar`, {});
  }

  buscarPorNombre(nombre: string): Observable<ResponseWrapper<Proveedor[]>> {
    return this.http.get<ResponseWrapper<Proveedor[]>>(`${this.baseUrl}/buscar?nombre=${nombre}`);
  }
}